sktime/forecasting/base/adapters/_fbprophet.py

```diff
@@ -116,11 +116,12 @@ def _fit(self, y, X, fh):
             df["cap"] = self.growth_cap
             df["floor"] = self.growth_floor
 
+        fit_kwargs = self.fit_kwargs or {}
```

Comment:
Is there any place where it's ensured that all subclasses will have this attribute? If not, this is going to fail with AttributeError. I'll suggest to use hasattr or getattr to be safe.

Reply:
yes, agreed - it also interacts with resolving the clash with the `sklearn` interface (write to `self` and do not change), so both should be addressed

Reply:
> @fkiraly could you elaborate on this one?

Not changing the init params written to self is a must, for `sklearn` interface compatibility.

What I was saying, @yarnabrina's request relates to the same param affected by my comment related to this requirement, `fit_params`, so whatever the change to address it, it should also address the `sklearn` compatibility issue.

Reply:
@yarnabrina Thanks for the input!

Please take a look at this draft PR on my fork: https://github.com/tpvasconcelos/sktime/pull/1

I think that this tackles all of the issues you mentioned. 

I had a look at your `_GeneralisedStatsForecastAdapter` and although I think this is a very good example where the adapter pattern makes sense (since you have at least 7 concrete forecaster implementations from this base), it does not apply to Prophet since prophet only has one implementation and the adapter class currently 100% depends on its concrete implementation. The pattern that you exemplified in `_GeneralisedStatsForecastAdapter` can also be seen in all other adaptors (pmdarima, pytorch, tbats, ...). Prophet is the ugly outlier here.

Reply:
@yarnabrina 

If you really don't like this approach and [really really want](https://www.youtube.com/watch?v=gJLIiF15wjQ) an adapter class for Prophet, then I would suggest doing the same thing that was done with the `_TbatsAdapter` and move all logic to the adapter class and only define `self._ModelClass` (via `_create_model_class()`) in the concrete implementation... But, for the reasons I mentioned above, I don't really see how this is a good pattern for Prophet and it would just add unnecessary complexity to an already quite large codebase 😬

Reply:
Take a look at #5592 please. That is a PR which is expecting the prophet adapter. Also, even if not in current sktime codebase (I'm not certain, I don't use it much myself) someone using sktime may have inherited it in their codebase and added custom functionalities. It's not a public class, and there's no promise of it henceforth, but I personally don't think removal of the adapter is necessary.

This is just my personal opinion as of now, and it'snot a very strong one. If you feel strongly that it's unnecessary, may be you can create a new design issue and ask for others opinions? You can ask on discord as well. Either way, removal will be beyond scope of this PR for sure.